src/compiler/checker.ts

```diff
@@ -17404,7 +17398,19 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isPatternLiteralPlaceholderType(type: Type): boolean {
-        return !!(type.flags & (TypeFlags.Any | TypeFlags.String | TypeFlags.Number | TypeFlags.BigInt)) || isPatternLiteralType(type);
+        if (type.flags & TypeFlags.Intersection) {
+            return some((type as IntersectionType).types, isPatternLiteralPlaceholderType);
+        }
+
+        if (type.flags & (TypeFlags.TemplateLiteral | TypeFlags.StringMapping)) {
+            return isPatternLiteralType(type);
+        }
+
+        if (type.flags & TypeFlags.ESSymbolLike) {
+            return false;
+        }
+
+        return !!(type.flags & (TypeFlags.Any | TypeFlags.Primitive));

```

Comment:
Then we can revert the rest to the original
```suggestion
return !!(type.flags & (TypeFlags.Any | TypeFlags.String | TypeFlags.Number | TypeFlags.BigInt)) || isPatternLiteralType(type);
```
so there's no surprising new behavior that doesn't involve intersections with literals in them.